socorro/processor/processor_2015.py

```diff
@@ -7,140 +7,158 @@
+    AuroraVersionFixitRule,
+    BetaVersionRule,
+    DatesAndTimesRule,
+    EnvironmentRule,
+    ESRVersionRewrite,
+    ExploitablityRule,
+    FlashVersionRule,
+    JavaProcessRule,
+    MissingSymbolsRule,
+    OSPrettyVersionRule,
+    OutOfMemoryBinaryRule,
+    PluginContentURL,
+    PluginRule,
+    PluginUserComment,
+    ProductRewrite,
+    ProductRule,
+    SignatureGeneratorRule,
+    ThemePrettyNameRule,
+    TopMostFilesRule,
+    UserDataRule,
+    Winsock_LSPRule,
+)
 
 
 class Processor2015(RequiredConfig):
     """this class is a generalization of the Processor into a rule processing
     framework. This class is suitable for use in the 'processor_app'
     introducted in 2012."""
 
-    required_config = Namespace()
+    required_config = Namespace('transform_rules')
+    required_config.add_option(
+        'database_class',
+        doc="the class of the database",
+        default='socorro.external.postgresql.connection_context.'
+                'ConnectionContext',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+                "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=str_to_python_object,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'dump_field',
+        doc='the default name of a dump',
+        default='upload_file_minidump',
+    )
+    required_config.command_pathname = change_default(
+        ExternalProcessRule,
+        'command_pathname',
+        # NOTE(willkg): This is the path for the RPM-based Socorro deploy. When
+        # we switch to Docker, we should change this.
+        '/data/socorro/stackwalk/bin/stackwalker',
+    )
+    required_config.add_option(
+        'result_key',
+        doc='the key where the external process result should be stored '
+            'in the processed crash',
+        default='%s_result' %
+            required_config.command_pathname.default.split('/')[-1]
+            .replace('-', ''),
+    )
     required_config.add_option(
-        name='rule_sets',
-        doc="a hierarchy of rules in json form",
-        default=default_rules_set_str,
-        from_string_converter=rule_sets_from_string,
-        likely_to_be_changed=True,
+        'return_code_key',
+        doc='the key where the external process return code should be stored '
+            'in the processed crash',
+        default='%s_return_code' %
+            required_config.command_pathname.default.split('/')[-1],
+    )
+    required_config.add_option(
+        name='symbols_urls',
+        doc='comma delimited ordered list of urls for symbol lookup',
+        default='https://localhost',
+        from_string_converter=str_to_list,
+        likely_to_be_changed=True
```

Comment:
I don't think I ever figured out what `likely_to_be_changed` does. What does it do?

Reply:
`likely_to_be_changed` is a flag indicating that this value should be written out when using configman to write out the configuration. With the flag present, configman writes the value as live configuration. Without it, the value is printed with a comment prefix.

Reply:
Ahhh... fascinating!